src/main/java/it/eng/idsa/businesslogic/processor/producer/websocket/client/MessageWebSocketOverHttpSender.java

```diff
@@ -0,0 +1,182 @@
+        return doSendMultipartMessageWebSocketOverHttps(header, payload, forwardTo, null);
+    }
+
+    public String sendMultipartMessageWebSocketOverHttps(String header, String payload, String forwardTo, Message message)
+            throws ParseException, IOException, KeyManagementException, NoSuchAlgorithmException, InterruptedException, ExecutionException {
+        return doSendMultipartMessageWebSocketOverHttps(header, payload, forwardTo, message);
+    }
+
+    private String doSendMultipartMessageWebSocketOverHttps(String header, String payload, String forwardTo, Message message)
+            throws ParseException, IOException, KeyManagementException, NoSuchAlgorithmException, InterruptedException, ExecutionException {
+        MultiPartMessage multipartMessage = new MultiPartMessage.Builder()
+                .setHeader(header)
+                .setPayload(payload)
+                .build();
+        extractWebSocketIPAndPort(forwardTo);
+        FileStreamingBean fileStreamingBean = webSocketClientConfiguration.fileStreamingWebSocket();
+        WebSocket wsClient = createWebSocketClient(message);
+        // Try to connect to the Server. Wait until you are not connected to the server.
+        fileStreamingBean.setup(wsClient);
+        fileStreamingBean.sendMultipartMessage(multipartMessage.toString());
+        //fileStreamingBean.sendMultipartMessage(multipartMessage);
+        // We don't have status of the response (is it 200 OK or not). We have only the content of the response.
+        String responseMessage = new String(webSocketClientConfiguration.responseMessageBufferWebSocketClient().remove());
+        closeWSClient(wsClient, message);
+        logger.info("received response: " + responseMessage);
+
+        return responseMessage;
+    }
+
+    @NotNull
+    private WebSocket createWebSocketClient(Message message) {
+        String WS_URL = "wss://" + webSocketHost + ":" + webSocketPort + HttpWebSocketServerBean.WS_URL;
+        WebSocket wsClient = null;
+        try {
+            final SslEngineFactory ssl = getSslEngineFactory();
+
+            DefaultAsyncHttpClientConfig clientConfig = new DefaultAsyncHttpClientConfig.Builder()
+                    .setDisableHttpsEndpointIdentificationAlgorithm(true)
+                    .setUseOpenSsl(true)
+                    .setSslEngineFactory(ssl)
+                    .build();
+
+            WebSocketUpgradeHandler.Builder upgradeHandlerBuilder
+                    = new WebSocketUpgradeHandler.Builder();
+            WebSocketUpgradeHandler wsHandler = upgradeHandlerBuilder
+                    .addWebSocketListener(webSocketClientConfiguration.inputStreamSocketListenerWebSocketClient()).build();
+            wsClient = asyncHttpClient(clientConfig)
+                    .prepareGet(WS_URL)
+                    .execute(wsHandler)
+                    .get();
+            return wsClient;
```

Comment:
- You can delete this line of the code (line 113) because, at the end of the method wsClient will be returned.

Reply:
Ok simple distraction error. Sorry